src/app/api/tickets/route.ts

```diff
@@ -0,0 +1,37 @@
+import { NextRequest, NextResponse } from "next/server";
+import { PrismaClient, RequestStatus } from "@prisma/client"; // Import RequestStatus enum
+
+const prisma = new PrismaClient(); // Initialize PrismaClient
+
+export async function GET(request: NextRequest) {
+  const { searchParams } = new URL(request.url);
+  const status = searchParams.get("status");
+
+  // Validate the status parameter
+  if (!status) {
+    return NextResponse.json({ error: "Missing status parameter" }, { status: 400 });
+  }
+
+  const validStatuses: RequestStatus[] = ["COMPLETED", "PENDING", "CANCELED"];
+  if (!validStatuses.includes(status as RequestStatus)) {
+    return NextResponse.json({ error: "Invalid status parameter" }, { status: 400 });
+  }
+
+  try {
+    const tickets = await prisma.residentRequest.findMany({
```

Comment:
Nitpick - why not residentRequests?
```suggestion
    const residentRequests = await prisma.residentRequest.findMany({
```